main.tf

```diff
@@ -63,23 +63,23 @@ module "tfe_workspace" {
   agent_role_arns                = var.tfe_workspace.agent_role_arns
   auth_method                    = var.tfe_workspace.auth_method
   auto_apply                     = var.tfe_workspace.auto_apply
-  branch                         = var.tfe_workspace.branch
+  branch                         = var.tfe_workspace.connect_vcs_repo != false ? var.tfe_workspace.branch : null
   clear_text_env_variables       = var.tfe_workspace.clear_text_env_variables
   clear_text_hcl_variables       = var.tfe_workspace.clear_text_hcl_variables
   clear_text_terraform_variables = merge(local.tfe_workspace.clear_text_terraform_variables, var.tfe_workspace.clear_text_terraform_variables)
   execution_mode                 = var.tfe_workspace.execution_mode
-  file_triggers_enabled          = var.tfe_workspace.file_triggers_enabled
+  file_triggers_enabled          = var.tfe_workspace.connect_vcs_repo != false ? var.tfe_workspace.file_triggers_enabled : null
   global_remote_state            = var.tfe_workspace.global_remote_state
   name                           = coalesce(var.tfe_workspace.name, var.name)
-  oauth_token_id                 = var.tfe_workspace.vcs_oauth_token_id
+  oauth_token_id                 = var.tfe_workspace.connect_vcs_repo != false ? var.tfe_workspace.vcs_oauth_token_id : null
   path                           = var.path
   permissions_boundary_arn       = try(aws_iam_policy.workspace_boundary[0].arn, null)
   policy                         = var.tfe_workspace.policy
   policy_arns                    = var.tfe_workspace.policy_arns
   project_id                     = var.tfe_workspace.project_id
   region                         = var.tfe_workspace.default_region
   remote_state_consumer_ids      = var.tfe_workspace.remote_state_consumer_ids
-  repository_identifier          = var.tfe_workspace.connect_vcs_repo ? var.tfe_workspace.repository_identifier : null
+  repository_identifier          = var.tfe_workspace.connect_vcs_repo != false ? coalesce(var.tfe_workspace.connect_vcs_repo, var.tfe_workspace.repository_identifier) : null
```

Comment:
can you explain this? is the current code `var.tfe_workspace.connect_vcs_repo ? var.tfe_workspace.repository_identifier : null` not already sufficient?